composeml/label_maker.py

```diff
@@ -60,7 +59,7 @@ def _preprocess(self, df):
 
         return df
 
-    def search(self, df, minimum_data, num_examples_per_instance, gap, verbose=True, *args, **kwargs):
+    def search(self, df, num_examples_per_instance, gap=1, minimum_data=0, verbose=True, *args, **kwargs):
```

Comment:
should gap default to `0`? basically, no gap between the end of the previous prediction window and start of the next?

Reply:
Gap is applied as an offset to the cutoff time to get the next cutoff time. If gap is 0, the cutoff time will remain the same for each example.

Reply:
for gap, I think it actually makes sense for it to default to `window_size`. that way we get training examples from non overlapping windows.

if `window_size` is `None`, I think gap can default to `1`. On that note, I think we also need to add support for a window size of `None`, which indicates that the window should be until the end of the data

Reply:
Currently, the last row overlaps when gap equals window size. I can update for consecutive non overlapping windows and add support for `window_size=None`.  Also if necessary, re-calibrate tests.

Can default value for `num_examples_per_instance=None` be all existing examples for that instance?

Reply:
can you explain more about what you mean by "Currently, the last row overlaps when gap equals window size"? 

I think default for `num_examples_per_instance`  could be just be one, or all possible examples. Not exactly sure which would be better. It's also possible we make that a required argument, so the user has to deal with it explicitly. That being said, how does a user indicate they want to make all possible examples. Can they do -1? If it's easier, perhaps we can create an issue for this question and discuss it there?

Reply:
The last row of a window is the first row for the next window. For example, given the following data:
###### Data
```
                     amount  customer_id
time                                    
2019-05-08 12:00:00       1            0
2019-05-08 12:05:00       1            0
2019-05-08 12:10:00       1            0
2019-05-08 12:15:00       1            0
2019-05-08 12:40:00       1            2
2019-05-08 12:45:00       1            2
```


When the window size and gap are equal, `window_size=2` and `gap=2`, this is the output for the first two data slices.

###### Slice (1)
```
                     amount  customer_id
time                                    
2019-05-08 12:00:00       1            0
2019-05-08 12:05:00       1            0
2019-05-08 12:10:00       1            0
```

In the second slice, the row with time `2019-05-08 12:10:00` occurs again.

###### Slice (2)
```
                     amount  customer_id
time                                    
2019-05-08 12:10:00       1            0
2019-05-08 12:15:00       1            0
```

I think the output for the first two data slices should be the following:

###### Slice (1)
```
                     amount  customer_id
time                                    
2019-05-08 12:00:00       1            0
2019-05-08 12:05:00       1            0
```
###### Slice (2)
```
                     amount  customer_id
time                                    
2019-05-08 12:10:00       1            0
2019-05-08 12:15:00       1            0
```

Reply:
For `num_examples_per_instance `, the user can use a large the number to get all possibles examples. It would be easier to use `-1` or `inf`. Yes, will create an issue for further discussion.

Reply:
yes, I think i agree with you. here's my understanding

the start of the `i + 1` slice should be the the start of the `i` slice + gap. so if things are zero indexed the starts should be 0, 2, 4. the window size then dicate the number of examples in the slice. here is the below with your example data


slice 1
```
                     amount  customer_id
time           
2019-05-08 12:00:00       1            0
2019-05-08 12:05:00       1            0
```

slice 2
```
                     amount  customer_id
time           
2019-05-08 12:10:00       1            0
2019-05-08 12:15:00       1            0
```

slice 3
```
                     amount  customer_id
time           
2019-05-08 12:40:00       1            2
2019-05-08 12:45:00       1            2
```

Reply:
Okay. Also, to confirm, the output would be the same as the three slices above if `window_size` and `gap` were both ten minutes.

Reply:
i think so, ya